src/main/java/bio/terra/service/auth/iam/IamResourceType.java

```diff
@@ -28,41 +32,32 @@ public String toString() {
     return samResourceName;
   }
 
+  @FunctionalInterface
+  private interface FilterFunction {
+    boolean apply(IamResourceType resourceType);
+  }
+
+  private static IamResourceType findIamResourceType(FilterFunction filter) {
+    return Arrays.stream(values()).filter(filter::apply).findFirst().orElse(null);
+  }
+
   @JsonCreator
   static IamResourceType fromValue(String text) {
-    for (IamResourceType b : IamResourceType.values()) {
-      if (StringUtils.equalsIgnoreCase(b.getSamResourceName(), text)) {
-        return b;
-      }
-    }
-    return null;
+    return findIamResourceType(
+        iamResourceType -> iamResourceType.getSamResourceName().equalsIgnoreCase(text));
   }
 
   public static IamResourceType fromEnum(IamResourceTypeEnum apiEnum) {
-    if (apiEnum == IamResourceTypeEnum.SPEND_PROFILE) {
-      return IamResourceType.SPEND_PROFILE;
-    }
-    return fromString(apiEnum.toString());
+    return findIamResourceType(iamResourceType -> iamResourceType.iamResourceTypeEnum == apiEnum);
   }
 
   public static IamResourceType fromString(String stringResourceType) {
-    for (IamResourceType b : IamResourceType.values()) {
-      if (b.getSamResourceName().equalsIgnoreCase(stringResourceType)) {
-        return b;
-      }
-    }
-    throw new RuntimeException("Invalid resource type: " + stringResourceType);
+    return findIamResourceType(
```

Comment:
So now this is not throwing, and before it did throw. If the caller isn't checking for `null` then it will cause an NPE later on. Either way the user gets a `500` response, the only difference is that checking here could produce a possibly more useful error message.

Reply:
Hmm... I wish these had consistent behavior, but I'll preserve the throwing behavior as it is right now for now and we can revisit that... eventually.